_commands/tools/grep.md

```diff
@@ -3,22 +3,137 @@
 
 grep
 -------
-TODO: Add documentation for this command by submitting a pull request.
-<!-- one line explanation would go here -->
 
-<!-- minimal example -->
+`grep` is a program for searching text files for lines that match regular exprssions. It can be used for all sorts of pattern-matching and text-based query analysis.
+
 ~~~ bash
-*short* example of command usage and output
+$ grep -i crab animals.txt
+Crab
+Crab-Eating Macaque
+Hermit Crab
+Horseshoe Crab
+King Crab
 ~~~
 
 <!--more-->
 
+This reference page uses the [animals.txt](/animals.txt) file for all of its examples. This list was taken from [Millie Bond's A-Z Index of Animals](https://a-z-animals.com/animals/).
```

Comment:
Would you mind moving the `animals.txt` file to `_commands/tools/src/animals.txt`? You can create this directory since it doesn't exist yet. That way in the future if other resources get added they can all go to the `src` directory and not crowd the main `.md` area.

Reply:
Done!